code/modules/clothing/head/soft_caps.dm

```diff
@@ -161,3 +162,40 @@
 /obj/item/clothing/head/soft/fishing_hat/Initialize(mapload)
 	. = ..()
 	AddElement(/datum/element/skill_reward, /datum/skill/fishing)
+
+/obj/item/clothing/head/soft/propeller_hat
+	name = "propeller hat"
+	desc = "A colorful hat with a spinning propeller sat on top."
+	icon_state = "propeller_hat"
+	soft_type = "propeller_hat"
+	inhand_icon_state = "rainbow_softcap"
+	worn_y_offset = 1
+	soft_suffix = null
+	actions_types = list(/datum/action/item_action/toggle)
+	var/enabled_waddle = TRUE
+	var/active = FALSE
+
+/obj/item/clothing/head/soft/propeller_hat/update_icon_state()
+	. = ..()
+	worn_icon_state = "[soft_type][flipped ? "_flipped" : null][active ? "_on" : null]"
+
+/obj/item/clothing/head/soft/propeller_hat/attack_self(mob/user)
+	active = !active
+	to_chat(user, span_notice("You [active ? "start" : "stop"] spinning the propeller."))
+	update_icon()
+	user.update_worn_head()
+	add_fingerprint(user)
+
+/obj/item/clothing/head/soft/propeller_hat/equipped(mob/living/user, slot)
+	. = ..()
+	if(slot & ITEM_SLOT_HEAD)
+		if(enabled_waddle)
+			user.AddElement(/datum/element/waddling)
```

Comment:
this is subtly broken in that elements dont stack. so if you have clown shoes on and a propeller hat, one will override the other.

this is stupid to fix and we need to have an actual system for it (and likely make it opt out rather than opt in). i would ignore it for now but im going to ping @LemonInTheDark and @Fikou very eye-rollily

Reply:
daedalus has a very simple system for it where some elements give you a trait and as long as you have it from at least one trait source it doesnt fully delete the element